CMakeLists.txt

```diff
@@ -11,11 +11,11 @@ set(R_INCLUDE_DIR ${R_HOME}/include)
 set(R_COMMAND ${R_HOME}/bin/R)
 
 add_definitions(-g3)
-set(CMAKE_CXX_FLAGS_RELEASE "-O2 -std=c++11")
-set(CMAKE_CXX_FLAGS_DEBUG "-O0 -Wall -std=c++11 -DENABLE_SLOWASSERT")
-set(CMAKE_CXX_FLAGS "-Wall -std=c++11 -fno-rtti")
-set(CMAKE_C_FLAGS_RELEASE "-O2 -std=gnu99")
-set(CMAKE_C_FLAGS_DEBUG "-Og -std=gnu99 -DENABLE_SLOWASSERT")
+set(CMAKE_CXX_FLAGS_RELEASE "-O2")
+set(CMAKE_CXX_FLAGS_DEBUG "-O0 -DENABLE_SLOWASSERT")
+set(CMAKE_CXX_FLAGS "-Wall -std=c++17 -fno-rtti")
```

Comment:
Well, if we're skipping R 3.4, might as well skip C++14!

I look forward to ~spamming~ using:

- generic lambdas
- s suffix for `std::string`
- `static_assert` without the message string
- nested namespace definitions
- static `if`
- `std::optional`